src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/RegisterUserTask.cs

```diff
@@ -85,6 +103,35 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
                     }
                     outcome = "Invalid";
                 }
+                else if (ConfirmUser)
+                {
+                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                    var request = _httpContextAccessor.HttpContext.Request;
+                    var uriBuilder = new UriBuilder(request.Scheme, request.Host.Host);
+                    if (request.Host.Port.HasValue)
+                        uriBuilder.Port = request.Host.Port.Value;
+                    uriBuilder.Path = string.Format("OrchardCore.Users/Registration/ConfirmEmail?userId={0}&code={1}", user.Id, code);
```

Comment:
Does the code already contain the user id?

Reply:
The UserManager.ConfirmEmailAsync requires a User and a token. This is how it is done in RegistrationController.ConfirmEmail action.